central/debug/service/zip.go

```diff
@@ -8,13 +8,20 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/stackrox/rox/pkg/k8sintrospect"
+	"github.com/stackrox/rox/pkg/sync"
 )
 
 var now = func() time.Time {
 	return time.Now()
 }
 
+var (
+	zipWriterMutex sync.Mutex
+)
+
 func writePrefixedFileToZip(zipWriter *zip.Writer, prefix string, file k8sintrospect.File) error {
+	zipWriterMutex.Lock()
```

Comment:
Yeah I should have done this sooner instead of leaving this "workaround" in, so did the following:
- new "zip writer" wrapper struct around the writer + guarding it with a mutext on a per-struct basis.

This makes things cleaner + resolves the issue with the parallel diag bundle requests.